File: front/src/api/uiTestSuiteService.js
import { http } from '@/utils/request'

export const UiTestSuiteApi = {
  // 获取UI测试套件列表
  getUiTestSuiteList: (params) => {
    // 构造干净的查询参数对象
    const cleanParams = {
      page: params.page || 1, // 默认值为 1
      pageSize: params.pageSize || 10, // 默认值为 10
    };
  
    // 仅在有值时添加其他参数
    if (params.keyword) {
      cleanParams.keyword = params.keyword;
    }
  
    if (params.status) {
      cleanParams.status = params.status;
    }

    // 添加项目ID参数（如果提供了的话）
    if (params.projectId) {
      cleanParams.projectId = params.projectId;
    }
  
    return http.get('/api/uitestsuites/list', { params: cleanParams });
  },
  
  // 获取UI测试套件详情
  getUiTestSuiteDetail: (id, projectId) => {
    if (!id) {
      return Promise.reject(new Error('ID不能为空'));
    }
    const params = {};
    if (projectId) params.projectId = projectId;
    
    return http.get(`/api/uitestsuites/${id}`, { params });
  },
  
  // 创建UI测试套件
  createUiTestSuite: (data) => {
    // 确保projectId是数字类型
    if (data.projectId && typeof data.projectId === 'string') {
      data.projectId = parseInt(data.projectId);
    }
    return http.post('/api/uitestsuites/create', data);
  },
  
  // 更新UI测试套件
  updateUiTestSuite: (id, data) => {
    if (!id) {
      return Promise.reject(new Error('ID不能为空'));
    }
    // 确保projectId是数字类型
    if (data.projectId && typeof data.projectId === 'string') {
      data.projectId = parseInt(data.projectId);
    }
    return http.put(`/api/uitestsuites/update/${id}`, data);
  },
  
  // 删除UI测试套件
  deleteUiTestSuite: (id, projectId) => {
    if (!id) {
      return Promise.reject(new Error('ID不能为空'));
    }
    const params = {};
    if (projectId) params.projectId = projectId;
    
    return http.delete(`/api/uitestsuites/delete/${id}`, { params });
  },
  
  // 执行UI测试套件
  executeUiTestSuite: (id, projectId) => {
    if (!id) {
      return Promise.reject(new Error('ID不能为空'));
    }
    const data = {};
    if (projectId) data.projectId = projectId;
    
    return http.post(`/api/uitestsuites/execute/${id}`, data);
  },
  
  // 获取套件中的测试用例
  getSuiteCases: (suiteId, projectId) => {
    if (!suiteId) {
      return Promise.reject(new Error('套件ID不能为空'));
    }
    const params = {};
    if (projectId) params.projectId = projectId;
    
    return http.get(`/api/uitestsuites/${suiteId}/cases`, { params });
  },
  
  // 更新套件中的测试用例
  updateSuiteCases: (suiteId, data) => {
    if (!suiteId) {
      return Promise.reject(new Error('套件ID不能为空'));
    }
    return http.put(`/api/uitestsuites/${suiteId}/cases`, data);
  },
  
  // 更新测试用例执行顺序
  updateCasesOrder: (suiteId, data) => {
    if (!suiteId) {
      return Promise.reject(new Error('套件ID不能为空'));
    }
    return http.put(`/api/uitestsuites/${suiteId}/cases-order`, data);
  },
  
  // 获取并发配置
  getConcurrencyConfig: (suiteId, projectId) => {
    if (!suiteId) {
      return Promise.reject(new Error('套件ID不能为空'));
    }
    const params = {};
    if (projectId) params.projectId = projectId;
    
    return http.get(`/api/uitestsuites/${suiteId}/concurrency`, { params });
  },
  
  // 更新并发配置
  updateConcurrencyConfig: (suiteId, data) => {
    if (!suiteId) {
      return Promise.reject(new Error('套件ID不能为空'));
    }
    return http.put(`/api/uitestsuites/${suiteId}/concurrency`, data);
  },
  
  // 获取定时任务配置
  getScheduleConfig: (suiteId, projectId) => {
    if (!suiteId) {
      return Promise.reject(new Error('套件ID不能为空'));
    }
    const params = {};
    if (projectId) params.projectId = projectId;
    
    return http.get(`/api/uitestsuites/${suiteId}/schedule`, { params });
  },
  
  // 创建或更新定时任务配置
  saveScheduleConfig: (suiteId, data) => {
    if (!suiteId) {
      return Promise.reject(new Error('套件ID不能为空'));
    }
    return http.post(`/api/uitestsuites/${suiteId}/schedule`, data);
  },
  
  // 删除定时任务配置
  deleteScheduleConfig: (suiteId, projectId) => {
    if (!suiteId) {
      return Promise.reject(new Error('套件ID不能为空'));
    }
    const params = {};
    if (projectId) params.projectId = projectId;
    
    return http.delete(`/api/uitestsuites/${suiteId}/schedule`, { params });
  }
}; 

File: verityx/src/main/java/com/example/verityx/service/UiTestSuiteService.java
package com.example.verityx.service;

import com.example.verityx.dto.PageResult;
import com.example.verityx.dto.UiTestSuiteDTO;

import java.util.List;

public interface UiTestSuiteService {
    
    /**
     * 分页获取UI测试套件列表
     *
     * @param page     页码
     * @param pageSize 每页大小
     * @param keyword  搜索关键词
     * @param status   套件状态
     * @param projectId 项目ID
     * @return 分页测试套件列表
     */
    PageResult<UiTestSuiteDTO> getUiTestSuiteList(Integer page, Integer pageSize, String keyword, String status, Long projectId);
    
    /**
     * 根据ID获取UI测试套件详情
     *
     * @param id 测试套件ID
     * @param projectId 项目ID
     * @return 测试套件详情
     */
    UiTestSuiteDTO getUiTestSuiteById(Long id, Long projectId);
    
    /**
     * 创建UI测试套件
     *
     * @param uiTestSuiteDTO 测试套件信息
     * @return 新创建的测试套件ID
     */
    Long createUiTestSuite(UiTestSuiteDTO uiTestSuiteDTO);
    
    /**
     * 更新UI测试套件
     *
     * @param id            测试套件ID
     * @param uiTestSuiteDTO 更新的测试套件信息
     */
    void updateUiTestSuite(Long id, UiTestSuiteDTO uiTestSuiteDTO);
    
    /**
     * 删除UI测试套件
     *
     * @param id 测试套件ID
     * @param projectId 项目ID
     */
    void deleteUiTestSuite(Long id, Long projectId);
    
    /**
     * 执行UI测试套件
     *
     * @param id 测试套件ID
     */
    void executeUiTestSuite(Long id);
    
    /**
     * 获取套件中的测试用例
     *
     * @param suiteId 测试套件ID
     * @return 测试用例列表
     */
    List<UiTestSuiteDTO.CaseInfo> getSuiteCases(Long suiteId);
    
    /**
     * 更新套件中的测试用例
     *
     * @param suiteId 测试套件ID
     * @param caseIds 测试用例ID列表
     */
    void updateSuiteCases(Long suiteId, List<Long> caseIds);
    
    /**
     * 更新测试套件的并发执行配置
     *
     * @param suiteId 测试套件ID
     * @param concurrencyConfig 并发配置信息
     */
    void updateConcurrencyConfig(Long suiteId, UiTestSuiteDTO.ConcurrencyConfig concurrencyConfig);
    
    /**
     * 获取测试套件的定时任务配置
     *
     * @param suiteId 测试套件ID
     * @return 定时任务配置信息
     */
    UiTestSuiteDTO.ScheduleConfig getScheduleConfig(Long suiteId);
    
    /**
     * 保存测试套件的定时任务配置
     *
     * @param suiteId 测试套件ID
     * @param scheduleConfig 定时任务配置信息
     */
    void saveScheduleConfig(Long suiteId, UiTestSuiteDTO.ScheduleConfig scheduleConfig);
    
    /**
     * 删除测试套件的定时任务配置
     *
     * @param suiteId 测试套件ID
     */
    void deleteScheduleConfig(Long suiteId);
} 